session.go

```diff
@@ -576,7 +576,43 @@ func (s *ProxySession) proxyFilterAppToTor() {
 						}
 						continue
 					} else {
-						log.Noticef("allowed ADD_ONION with %s", cmdLine)
+						if s.policy.OzForwardOnion == true {
+							if s.policy.OzApp == "" {
+								log.Errorf("Missing Oz profile name, filter policy syntax error on %s so DENY: %s A->T: [%s]\n", s.policy.ExecPath, appName, cmdLine)
+								_, err = s.appConnWrite(false, []byte("250 Tor Control command proxy denied: filtration policy.\r\n"))
+								if err != nil {
+									s.errChan <- err
+								}
+								continue
+							}
+							id, err := s.findOzSandbox(s.policy.OzApp)
+							if err != nil {
+								log.Errorf("Could not lookup %s sandbox ID for %s so DENY: %s A->T: [%s]\n", s.policy.OzApp, s.policy.ExecPath, appName, cmdLine)
+								_, err = s.appConnWrite(false, []byte("250 Tor Control command proxy denied: filtration policy.\r\n"))
+								if err != nil {
+									s.errChan <- err
+								}
+								continue
+							}
+							keytype, keyblob, onionPort, localPort, err := s.dissectOnion(cmdLine)
```

Comment:
the error condition of s.dissectOnion needs to be checked here before we overwrite err